src/test/java/io/vertx/core/http/HttpTest.java

```diff
@@ -782,6 +782,29 @@ public void testNoParams() {
     await();
   }
 
+  @Test
+  public void testGetParamDefaultValue() {
+    String paramName1 = "foo";
+    String paramName1Value = "bar";
+    String paramName2 = "notPresentParam";
+    String defaultValue = "defaultValue";
+    String reqUri = String.format("%s/?%s=%s", DEFAULT_TEST_URI, paramName1, paramName1Value);
+
+    server.requestHandler(req -> {
+      assertTrue(req.params().contains(paramName1));
+      assertEquals(paramName1Value, req.getParam(paramName1, defaultValue));
+      assertFalse(req.params().contains(paramName2));
+      assertEquals(defaultValue, req.getParam(paramName2, defaultValue));
+      req.response().end();
+    });
+
+    server.listen(onSuccess(server -> {
```

Comment:
please start server like in other tests (using testAddress) so the test can run with other config such as native transport.